### PR TITLE
docs/ci: use myst, fix broken links, add linkcheck test, remove deprecated distutils, avoid 2x job triggers

### LINKED ISSUE #480: Drop dependency on recommonmark - use MyST instead?
We have a dependency on recommonmark for our docs, but that makes us not able to work with Python 3.9 because its dependency commonmark hasn't updated to support Python 3.9.

I'd love to transition to have MyST instead. I think perhaps we use recommonmark to have markdown files part of Sphinx, while we could transition to use MyST instead.

### PR DESCRIPTION
**Pull request by consideRatio**
The docs had not been updated for a long time, so this PR is a significant overhaul. There were plenty of broken links etc.

Closes #480.

There is one thing that doesn't fit so well in this PR, which is [72ce042](https://github.com/jupyterhub/oauthenticator/pull/511/commits/72ce042c0812dc1be0755aaec36761e3d6cbe15a) that was added because I could not get `make html` or `make linkcheck` to function properly without making some changes there. Closes #505 which was a dedicated PR trying to do something about that as well.

### CODE REVIEW

File: docs/source/getting-started.rst [hunk 1]
```diff
@@ -287,6 +283,13 @@ You can customize the sign in button text (optional):
 
    c.GoogleOAuthenticator.login_service = 'My College'
 
+.. note::
+
+   Additional notes, that seem quite outdated at the time of writing May 2022,
+   are available about authorizing users part of specific Google Groups are
+   :ref:`available here <google-groups-label>`. Contributions to update these
+   and re-verify this functionality are most welcome.
+
```
Comment (GeorgianaElena) [inline docs/source/getting-started.rst:292]: Oh yes!  👀

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -5,7 +5,19 @@ name: Tests
 
 on:
   pull_request:
+    paths-ignore:
+      - "docs/**"
+      - ".github/workflows/*.yaml"
+      - "!.github/workflows/publish.yaml"
```
Comment (GeorgianaElena) [inline .github/workflows/test.yml:11]: Any reason not to ignore the publish workflow here?

```suggestion
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -5,7 +5,19 @@ name: Tests
...
   push:
+    paths-ignore:
+      - "docs/**"
+      - ".github/workflows/*.yaml"
+      - "!.github/workflows/publish.yaml"
```
Comment (GeorgianaElena) [inline .github/workflows/test.yml:16]: Same

```suggestion
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (consideRatio) [reply .github/workflows/test.yml]: Nice catch! It was copy pasted from a workflow named publish, but I had forgo to update it to de-ignore the `test.yaml` workflow. The idea was that "ignore all workflows (line before), but not this workflow"

**Comment by consideRatio** [comment]
@GeorgianaElena thank you for the review!!! I applied your suggestions, I'll go for a merge right now then!

**Pull request merged**